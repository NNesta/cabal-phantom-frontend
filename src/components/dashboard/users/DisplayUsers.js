import React, { Component } from 'react';
import { getUsersAction } from '../../../redux/actions/getUsersAction'
import { Table } from 'reactstrap'
import BorderColorIcon from '@material-ui/icons/BorderColor';
import DeleteIcon from '@material-ui/icons/Delete';
import { connect } from "react-redux";
import AddRoundedIcon from '@material-ui/icons/AddRounded';
import ContentHeader from '../helpers/ContentHeader'
import '../../../assets/style/dashboard.scss'
import AddNewUser from './AddNewUser'
import Skeleton from "react-loading-skeleton";


class DisplayUsers extends Component {

    constructor() {
        super();
        this.state = {
            showHide: false,
            bgColor: ''
        }
    }

    handleModalShowHide() {
        this.setState({ showHide: !this.state.showHide })
    }

    componentDidMount() {
        this.props.getUsersAction()
    }

    render() {
        return (
            <div className="p-4">
                <ContentHeader
                    title="Users"
                    Icon={AddRoundedIcon}
                    name="Add"
                    onClick={() => this.handleModalShowHide()}
                />

                <Table bordered striped className="h4 table-responsive-sm">
                    <thead>
                        <tr>
                            <th>ID</th>
                            <th>Name</th>
                            <th>Email</th>
                            <th>Role</th>
                            <th colSpan={2}>Actions</th>
                        </tr>
                    </thead>
                        {this.props.getUsers.loading === 'none' ? (
                                <tbody>
                                    {this.props.getUsers.items.data.users.map((user) => {
                                        return ( 
                                        <tr key={user.id}>
                                            <th scope="row">{user.id}</th>
                                            <td>{user.name}</td>
                                            <td>{user.email}</td>
                                            <td>{user.role}</td>
                                            <td style={{ color: 'blue', fontSize: '1.8rem' }}><BorderColorIcon /></td>
                                            <td style={{ color: 'red', fontSize: '1.8rem' }}><DeleteIcon /></td>
                                        </tr>
                                        )
                                })} 
                                </tbody>
                            ) : (
                                <tbody>
                                    {[1, 2, 3, 4, 5, 6, 7, 8, 9, 10].map((user) => {
                                        return (
                                            <tr key={user}>
                                                <th scope="row"><Skeleton /></th>
                                                <td><Skeleton /></td>
                                                <td><Skeleton /></td>
                                                <td><Skeleton /></td>
                                                <td style={{ color: 'blue', fontSize: '1.8rem' }}><BorderColorIcon /></td>
                                                <td style={{ color: 'red', fontSize: '1.8rem' }}><DeleteIcon /></td>
                                            </tr>
                                        )
                                    })}
                                </tbody>
                            )
                            }

                </Table>
                <AddNewUser
                    isOpen={this.state.showHide}
                    onClick={() => this.handleModalShowHide()}
                />
            </div>
        );
    }
}

const mapStateToProps = (state) => {
    return {
        getUsers: state.getUsers
    };
};

export default connect(mapStateToProps, { getUsersAction })(DisplayUsers);

