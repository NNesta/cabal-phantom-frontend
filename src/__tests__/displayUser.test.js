import DisplayUsers from '../components/dashboard/users/DisplayUsers';
import React from 'react';
import { render, fireEvent, cleanup, waitFor } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/store';
import userEvent from '@testing-library/user-event';
import { BrowserRouter as Router, Link } from 'react-router-dom';


const HomeComponent = () => {
	return render(
		<Provider store={store}>
				<Router>
					<DisplayUsers />
				</Router>
		</Provider>
	);
};

describe('Home component', () => {
    afterEach(cleanup)

    it('Should render home component properly', () => {
		const { asFragment } = HomeComponent();
		expect(asFragment(<DisplayUsers />)).toMatchSnapshot();
	});
})