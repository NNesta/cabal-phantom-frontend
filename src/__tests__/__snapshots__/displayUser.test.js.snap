// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component Should render home component properly 1`] = `
<DocumentFragment>
  <div
    class="p-4"
  >
    <div
      class="d-flex justify-content-between align-items-center"
      style="padding: 1rem 0px; border-bottom: 2px solid #fff; font-weight: bolder;"
    >
      <h2
        class="text-primary"
      >
        Users
      </h2>
      <div
        class="button_icon"
      >
        <button
          class="d-flex align-items-center btn btn-outline-primary"
          type="button"
        >
          <div>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              style="font-size: 2rem;"
              viewBox="0 0 24 24"
            >
              <path
                d="M18 13h-5v5c0 .55-.45 1-1 1s-1-.45-1-1v-5H6c-.55 0-1-.45-1-1s.45-1 1-1h5V6c0-.55.45-1 1-1s1 .45 1 1v5h5c.55 0 1 .45 1 1s-.45 1-1 1z"
              />
            </svg>
          </div>
          <div
            class="ml-2"
          >
            Add
          </div>
        </button>
      </div>
    </div>
    <table
      class="h4 table-responsive-sm table table-bordered table-striped"
    >
      <thead>
        <tr>
          <th>
            ID
          </th>
          <th>
            Name
          </th>
          <th>
            Email
          </th>
          <th>
            Role
          </th>
          <th
            colspan="2"
          >
            Actions
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <th
            scope="row"
          >
            <span>
              <span
                class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
              >
                ‌
              </span>
            </span>
          </th>
          <td>
            <span>
              <span
                class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
              >
                ‌
              </span>
            </span>
          </td>
          <td>
            <span>
              <span
                class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
              >
                ‌
              </span>
            </span>
          </td>
          <td>
            <span>
              <span
                class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
              >
                ‌
              </span>
            </span>
          </td>
          <td
            style="color: blue; font-size: 1.8rem;"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M17.75 7L14 3.25l-10 10V17h3.75l10-10zm2.96-2.96c.39-.39.39-1.02 0-1.41L18.37.29a.9959.9959 0 0 0-1.41 0L15 2.25 18.75 6l1.96-1.96z"
              />
              <path
                d="M0 20h24v4H0z"
                fill-opacity=".36"
              />
            </svg>
          </td>
          <td
            style="color: red; font-size: 1.8rem;"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
          </td>
        </tr>
        <tr>
          <th
            scope="row"
          >
            <span>
              <span
                class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
              >
                ‌
              </span>
            </span>
          </th>
          <td>
            <span>
              <span
                class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
              >
                ‌
              </span>
            </span>
          </td>
          <td>
            <span>
              <span
                class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
              >
                ‌
              </span>
            </span>
          </td>
          <td>
            <span>
              <span
                class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
              >
                ‌
              </span>
            </span>
          </td>
          <td
            style="color: blue; font-size: 1.8rem;"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M17.75 7L14 3.25l-10 10V17h3.75l10-10zm2.96-2.96c.39-.39.39-1.02 0-1.41L18.37.29a.9959.9959 0 0 0-1.41 0L15 2.25 18.75 6l1.96-1.96z"
              />
              <path
                d="M0 20h24v4H0z"
                fill-opacity=".36"
              />
            </svg>
          </td>
          <td
            style="color: red; font-size: 1.8rem;"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
          </td>
        </tr>
        <tr>
          <th
            scope="row"
          >
            <span>
              <span
                class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
              >
                ‌
              </span>
            </span>
          </th>
          <td>
            <span>
              <span
                class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
              >
                ‌
              </span>
            </span>
          </td>
          <td>
            <span>
              <span
                class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
              >
                ‌
              </span>
            </span>
          </td>
          <td>
            <span>
              <span
                class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
              >
                ‌
              </span>
            </span>
          </td>
          <td
            style="color: blue; font-size: 1.8rem;"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M17.75 7L14 3.25l-10 10V17h3.75l10-10zm2.96-2.96c.39-.39.39-1.02 0-1.41L18.37.29a.9959.9959 0 0 0-1.41 0L15 2.25 18.75 6l1.96-1.96z"
              />
              <path
                d="M0 20h24v4H0z"
                fill-opacity=".36"
              />
            </svg>
          </td>
          <td
            style="color: red; font-size: 1.8rem;"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
          </td>
        </tr>
        <tr>
          <th
            scope="row"
          >
            <span>
              <span
                class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
              >
                ‌
              </span>
            </span>
          </th>
          <td>
            <span>
              <span
                class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
              >
                ‌
              </span>
            </span>
          </td>
          <td>
            <span>
              <span
                class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
              >
                ‌
              </span>
            </span>
          </td>
          <td>
            <span>
              <span
                class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
              >
                ‌
              </span>
            </span>
          </td>
          <td
            style="color: blue; font-size: 1.8rem;"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M17.75 7L14 3.25l-10 10V17h3.75l10-10zm2.96-2.96c.39-.39.39-1.02 0-1.41L18.37.29a.9959.9959 0 0 0-1.41 0L15 2.25 18.75 6l1.96-1.96z"
              />
              <path
                d="M0 20h24v4H0z"
                fill-opacity=".36"
              />
            </svg>
          </td>
          <td
            style="color: red; font-size: 1.8rem;"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
          </td>
        </tr>
        <tr>
          <th
            scope="row"
          >
            <span>
              <span
                class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
              >
                ‌
              </span>
            </span>
          </th>
          <td>
            <span>
              <span
                class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
              >
                ‌
              </span>
            </span>
          </td>
          <td>
            <span>
              <span
                class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
              >
                ‌
              </span>
            </span>
          </td>
          <td>
            <span>
              <span
                class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
              >
                ‌
              </span>
            </span>
          </td>
          <td
            style="color: blue; font-size: 1.8rem;"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M17.75 7L14 3.25l-10 10V17h3.75l10-10zm2.96-2.96c.39-.39.39-1.02 0-1.41L18.37.29a.9959.9959 0 0 0-1.41 0L15 2.25 18.75 6l1.96-1.96z"
              />
              <path
                d="M0 20h24v4H0z"
                fill-opacity=".36"
              />
            </svg>
          </td>
          <td
            style="color: red; font-size: 1.8rem;"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
          </td>
        </tr>
        <tr>
          <th
            scope="row"
          >
            <span>
              <span
                class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
              >
                ‌
              </span>
            </span>
          </th>
          <td>
            <span>
              <span
                class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
              >
                ‌
              </span>
            </span>
          </td>
          <td>
            <span>
              <span
                class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
              >
                ‌
              </span>
            </span>
          </td>
          <td>
            <span>
              <span
                class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
              >
                ‌
              </span>
            </span>
          </td>
          <td
            style="color: blue; font-size: 1.8rem;"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M17.75 7L14 3.25l-10 10V17h3.75l10-10zm2.96-2.96c.39-.39.39-1.02 0-1.41L18.37.29a.9959.9959 0 0 0-1.41 0L15 2.25 18.75 6l1.96-1.96z"
              />
              <path
                d="M0 20h24v4H0z"
                fill-opacity=".36"
              />
            </svg>
          </td>
          <td
            style="color: red; font-size: 1.8rem;"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
          </td>
        </tr>
        <tr>
          <th
            scope="row"
          >
            <span>
              <span
                class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
              >
                ‌
              </span>
            </span>
          </th>
          <td>
            <span>
              <span
                class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
              >
                ‌
              </span>
            </span>
          </td>
          <td>
            <span>
              <span
                class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
              >
                ‌
              </span>
            </span>
          </td>
          <td>
            <span>
              <span
                class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
              >
                ‌
              </span>
            </span>
          </td>
          <td
            style="color: blue; font-size: 1.8rem;"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M17.75 7L14 3.25l-10 10V17h3.75l10-10zm2.96-2.96c.39-.39.39-1.02 0-1.41L18.37.29a.9959.9959 0 0 0-1.41 0L15 2.25 18.75 6l1.96-1.96z"
              />
              <path
                d="M0 20h24v4H0z"
                fill-opacity=".36"
              />
            </svg>
          </td>
          <td
            style="color: red; font-size: 1.8rem;"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
          </td>
        </tr>
        <tr>
          <th
            scope="row"
          >
            <span>
              <span
                class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
              >
                ‌
              </span>
            </span>
          </th>
          <td>
            <span>
              <span
                class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
              >
                ‌
              </span>
            </span>
          </td>
          <td>
            <span>
              <span
                class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
              >
                ‌
              </span>
            </span>
          </td>
          <td>
            <span>
              <span
                class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
              >
                ‌
              </span>
            </span>
          </td>
          <td
            style="color: blue; font-size: 1.8rem;"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M17.75 7L14 3.25l-10 10V17h3.75l10-10zm2.96-2.96c.39-.39.39-1.02 0-1.41L18.37.29a.9959.9959 0 0 0-1.41 0L15 2.25 18.75 6l1.96-1.96z"
              />
              <path
                d="M0 20h24v4H0z"
                fill-opacity=".36"
              />
            </svg>
          </td>
          <td
            style="color: red; font-size: 1.8rem;"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
          </td>
        </tr>
        <tr>
          <th
            scope="row"
          >
            <span>
              <span
                class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
              >
                ‌
              </span>
            </span>
          </th>
          <td>
            <span>
              <span
                class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
              >
                ‌
              </span>
            </span>
          </td>
          <td>
            <span>
              <span
                class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
              >
                ‌
              </span>
            </span>
          </td>
          <td>
            <span>
              <span
                class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
              >
                ‌
              </span>
            </span>
          </td>
          <td
            style="color: blue; font-size: 1.8rem;"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M17.75 7L14 3.25l-10 10V17h3.75l10-10zm2.96-2.96c.39-.39.39-1.02 0-1.41L18.37.29a.9959.9959 0 0 0-1.41 0L15 2.25 18.75 6l1.96-1.96z"
              />
              <path
                d="M0 20h24v4H0z"
                fill-opacity=".36"
              />
            </svg>
          </td>
          <td
            style="color: red; font-size: 1.8rem;"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
          </td>
        </tr>
        <tr>
          <th
            scope="row"
          >
            <span>
              <span
                class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
              >
                ‌
              </span>
            </span>
          </th>
          <td>
            <span>
              <span
                class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
              >
                ‌
              </span>
            </span>
          </td>
          <td>
            <span>
              <span
                class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
              >
                ‌
              </span>
            </span>
          </td>
          <td>
            <span>
              <span
                class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
              >
                ‌
              </span>
            </span>
          </td>
          <td
            style="color: blue; font-size: 1.8rem;"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M17.75 7L14 3.25l-10 10V17h3.75l10-10zm2.96-2.96c.39-.39.39-1.02 0-1.41L18.37.29a.9959.9959 0 0 0-1.41 0L15 2.25 18.75 6l1.96-1.96z"
              />
              <path
                d="M0 20h24v4H0z"
                fill-opacity=".36"
              />
            </svg>
          </td>
          <td
            style="color: red; font-size: 1.8rem;"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
          </td>
        </tr>
      </tbody>
    </table>
    <div />
  </div>
</DocumentFragment>
`;
